submissions/shramkoweb/friends-app/script.js

```diff
@@ -0,0 +1,168 @@
+};
+
+const handleResetClick = () => {
+    state.search = '';
+    state.age = null;
+    state.name = null;
+    state.gender = Gender.ALL;
+
+    render(state);
+};
+
+const filterByGender = (users, gender) => {
+    if (gender === Gender.ALL) {
+        return users;
+    }
+
+    return users.filter(user => user.gender === gender);
+};
+
+const filterByQuery = (users, query) => {
+    if (query === '') {
+        return users;
+    }
+
+    return users.filter(
+        user => user.name.first
+            .toLowerCase()
+            .includes(query)
+    );
+};
+
+const sortByName = (users, sortType) => {
+    switch (sortType) {
+        case 'a':
+            return users.sort((a, b) => {
+                const firstNAme = a.name.first.toUpperCase();
+                const secondName = b.name.first.toUpperCase();
+
+                if (firstNAme < secondName) {
+                    return -1;
+                }
+                if (firstNAme > secondName) {
+                    return 1;
+                }
+
+                return 0;
+            });
+        case 'z':
+            return users.sort((a, b) => {
+                const firstNAme = a.name.first.toUpperCase();
+                const secondName = b.name.first.toUpperCase();
+
+                if (firstNAme < secondName) {
+                    return 1;
+                }
+                if (firstNAme > secondName) {
+                    return -1;
+                }
+
+                return 0;
+            });
+        default:
+            return users;
+    }
+};
+
+const sortBy = (users, sortType) => {
+    switch (sortType) {
+        case SortType.ASCENDING:
+            return users.sort((a, b) => a.dob.age - b.dob.age);
+        case SortType.DESCENDING:
+            return users.sort((a, b) => b.dob.age - a.dob.age);
+        default:
+            return users;
+    }
+};
+
+const renderCard = (data) => {
+    const { name, dob, gender, email, picture } = data;
+
+    const markup = `
+        <li class="friends__item">
+          <section class="friend-card">
+            <div class="friend-card__image-wrapper">
+              <img class="friend-card__image" src=${picture.large} alt=${name.first}>
+            </div>
+            <h3 class="friend-card__title">${name.first} ${name.last}</h3>
+            <a class="friend-card__mail" href="mailto:${email}">${email}</a>        
+            <p class="friend-card__sex">Gender - ${gender}</p>
+            <p class="friend-card__age">Age - ${dob.age}</p>
+          </section>
+        </li>
+    `;
+
+    return createElement(markup);
+};
+
+const renderCards = (cards, target) => {
+    const fragment = document.createDocumentFragment();
+    cards.forEach(card => fragment.appendChild(renderCard(card)));
+
+    target.appendChild(fragment);
+};
+
+const render = (state) => {
+    listElement.innerHTML = '';
+    const sorted = sortBy(sortByName(state.users, state.name), state.age);
```

Comment:
You sorting by name always after sorting by asc-desc. It is doesn't work correctly, i think. What if i want sort by asc or desc first of all? What you think about it?

Reply:
There are 2 types of sorting in this application. By name and age, so they will always be applied in one order or another.
If I have both sorting enabled, what do you suggest turning off the other when you remove one of them? It's a bad pattern to change something instead of a user.

Reply:
If you need only one sort of sorting, reset the form and select this type separately. I do not see any other way out yet, if there are suggestions I will listen.